### ISSUE TITLE
Write More Doc Strings

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#11159]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#1604|
|Type|enhancement|
|Created|2006-03-25 18:10:28Z|

http://twistedmatrix.com/pipermail/twisted-python/2006-March/012736.html

<details><summary>Searchable metadata</summary>

```
trac-id__1604 1604
type__enhancement enhancement
reporter__exarkun exarkun
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__documentation documentation
time__1143310228000000 1143310228000000
changetime__1149901910000000 1149901910000000
version__ 
owner__everybody everybody

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set owner to everybody|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun commented|
|-|-|

See UndocumentedTwisted for progress on this ticket.

**Comment by twisted-trac** [comment]
|[<img alt="dreid's avatar" src="https://avatars.githubusercontent.com/u/48695?s=50" width="50" height="50">](https://github.com/dreid)|@dreid commented|
|-|-|

branched to [branches/write-more-doc-strings-1604](branches/write-more-doc-strings-1604) at r16442

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set status to `closed`|
|-|-|

(In [17154]) Merge write-more-doc-strings-1604

Author: jml
Reviewer: exarkun
Fixes [#11159](https://github.com/twisted/twisted/issues/11159)

Adds a lot of docstrings to various trial classes and interfaces.  The complete
task outlined for this ticket has not been completed, but no changes have been
made in this branch in quite some time and it seems unlikely more will be 
forthcoming.
